modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -7,6 +8,22 @@
 include_once 'manatime_feature_timesheets.features.inc';
 
 /**
+ * Implements hook_menu().
+ */
+function manatime_feature_timesheets_menu() {
+  $items = array();
+
+  $items['timesheet/add-new-form/%/%ctools_js'] = array(
+    'page callback' => 'manatime_feature_timesheets_add_row_callback',
+    'page arguments' => array(2, 3),
+    'access callback' => 'user_is_logged_in',
```

Comment:
Let me show you a trick :grin:

If you change path to "timesheet/add-new-form/%node/%ctools_js" you will get the node object loaded instead of the nid; so, for access callback you could create a custom function, let's say "manatime_feature_timesheet_validate_user" and you receive the node, then, you load user from `global $user` and `return $node->uid === $user->uid;`. Let's talk via chat if you need more help on this.